### PR TITLE
Pre multiplayer

### PR DESCRIPTION
**Pull request by Taeir**
**TL;DR: Don't add variables in Serializable classes.**
# Pre multiplayer changes

This pull request contains code changes that are neccessary for multiplayer.
However, this pull request does not contain any multiplayer code.
## A small overview of the changes:
- Made many classes serializable, and added writeObject and readObject methods where needed. These methods are used by the serialization (**but do not show up as being called**), which will be used to send objects in a multiplayer connection.
- Added entityID's. EntityID's will be used by multiplayer.
- Changed EnemyFishFactory#randomizedFish code to be able to work with multiple areas at once.
- Made some Logger and Settings instances protected, and some protected static. This decreases the amount of settings objects in the code.
- Moved GameThread up from PlayingField to it's subclasses. Multiplayer will use a different GameThread, so this change is neccesary. (8973c54)
- Added getController methods (8a9ab4d)
- KeyListenerBehaviour now registers on a scene, not a stage. This prevents a few potential problems. (36c49d7)
- KeyListenerBehaviour can now be unregistered. (36c49d7)
- Added updateTo methods in IMoveBehaviour and ICollisionArea. These will be used by the multiplayer implementation to update the client side state to the server side state. (5a4bbb2 and c079dff)

The serializable change has some consequences:
**Adding a variable to a serializable class that does not have readObject/writeObject methods WILL cause multiplayer problems. The field you are adding WILL be sent to the client, and if java does not know how, we WILL get exceptions.**
## Reviewing:

This pull request is a bit large (that's why I'm splitting multiplayer into multiple pull requests in the first place), but I think most changes are easy to understand. Just take a look at it, there is no particular order required, because this pull request consists of multiple smaller changes that are not neccesarily related.

<organisation, interface, feature, layout

### CODE REVIEW

File: src/main/java/com/github/fishio/SinglePlayerPlayingField.java [hunk 1]
```diff
@@ -25,8 +32,13 @@
 	 * @param canvas
 	 *            the canvas to use, can be <code>null</code> to create one.
 	 */
-	public SinglePlayerPlayingField(int fps, Canvas canvas) {
```
Comment (Diocruel) [inline src/main/java/com/github/fishio/SinglePlayerPlayingField.java:28]: _Auto-generated comment_

```
Expected @param tag for 'scene'.
```

Checkstyle warning, but a let's not worry about this now!
<layout

File: src/main/java/com/github/fishio/behaviours/KeyListenerBehaviour.java [hunk 2]
```diff
@@ -1,7 +1,16 @@
...
+
+import com.github.fishio.PlayerFish;
 import com.github.fishio.Vec2d;
+import com.github.fishio.settings.Settings;
 
+import javafx.event.EventHandler;
+import javafx.scene.Scene;
 import javafx.scene.input.KeyCode;
 import javafx.scene.input.KeyEvent;
 import javafx.stage.Stage;
```
Comment (Diocruel) [inline src/main/java/com/github/fishio/behaviours/KeyListenerBehaviour.java:16]: _Auto-generated comment_

```
Avoid unused imports such as 'javafx.stage.Stage'
```

<layout

**Comment by Diocruel** [comment]
Made some comments about the static analysis tools, but for the rest it seems to be a solid start for multi player. I suggest not fixing these remarks and continuing with the implementation, which is really needed to meet the deadline!
+1

<review, layout

**Comment by ChielBruin** [comment]
Looks good!
I agree with @Diocruel that the suggested changes should be fixed later
When the merge conflict is resolved it is ready to merge
<review

**Comment by jessearens** [comment]
Even though I think I only understand half of this, it seems good to me.

<feature

**Comment by Taeir** [comment]
@Diocruel @ChielBruin I implemented the suggested changes, but on a different branch with a lot of other changes. I suggest merging this now and having those changes added in the next pull request.

<review
